python/cudf/cudf/core/dataframe.py

```diff
@@ -1580,7 +1580,7 @@ def _take_columns(self, positions):
             else:
                 colname = columns[positions[idx]]
             if len(self) == 0:
-                result[colname] = as_column([])
+                result[colname] = as_column([], dtype=self.dtypes[idx])
```

Comment:
Looks like this is somehow causing a couple of test failures in cudf tests where things are becoming `object` type instead of maintaining `float64` type.

Reply:
Interesting - Just looked into this.  For these tests we are using `iloc` to select both columns and rows.  In cudf, we first select the columns, then select the rows.  For the failing cases, we start by selecting an empty slice of columns (resulting in an `object` dtype, since we don't actually have any columns to determine the types).  Since we are no longer using `float64` as the default for empty data during the follow-up row selection, we get different behavior than pandas (and thus the failures).

In cases like this (when we have an empty-column selection), should we just set the dtype to `"float64"`?  This is an easy change, and is consistent with pandas.

Reply:
Grr - Changing the default typ for empty column selection messed up all the `repr` tests, so I am just letting the default dtype take over during row selection when the original type is `"object"` - This should pass tests, but may not be ideal